src/relax/transform/canonicalize_bindings.cc

```diff
@@ -91,18 +91,20 @@ class CanonicalizePlanner : public ExprVisitor {
         bound_to = opt.value();
       }
 
-      if (bound_var.as<DataflowVarNode>() || !bound_to.as<DataflowVarNode>()) {
+      if (bound_var.as<DataflowVarNode>() || !bound_to.as<DataflowVarNode>() ||
+          !visitor.used_outside_home_dataflow_.count(bound_var)) {
         // Case 1: Var = Var
         // Case 2: DataflowVar = Var
         // Case 3: DataflowVar = DataflowVar
+        // Case 4a: Var = DataflowVar, but used outside this DataflowBlock
```

Comment:
```suggestion
        // Case 4a: Var = DataflowVar, but not used outside this DataflowBlock
```
It appears the text was copied from case 4b. I also think it would be clearer to write out that it is the var that is not used outside the DF block

Reply:
Thank you, and updated the comment to be more explicit.  I've also changed "this DataflowBlock" to "the DataflowBlock containing the binding", since this function is called after the entire function is visited, not during the visit of any specific dataflow block.